src/ORM/UnionList.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+namespace SilverStripe\ORM;
+
+use SilverStripe\Dev\Debug;
+use SilverStripe\View\ViewableData;
+use LogicException;
+
+/**
+ * A list object that wraps around other SS_List instances.
+ */
+class UnionList extends ViewableData implements SS_List
+{
+    /**
+     * An array of SS_List instances
+     *
+     * @var array
+     */
+    protected $lists = array();
+
+    /**
+     * Create a new UnionList.
+     *
+     * @param array $lists - Lists to hold
+     */
+    public function __construct(array $lists)
+    {
+        foreach ($lists as $list) {
+            $this->lists[] = clone $list;
+        }
+        parent::__construct();
+    }
+
+    /**
+     * Return the total number of items in each SS_List.
+     *
+     * @return int
+     */
+    public function count()
+    {
+        $count = 0;
+        foreach ($this->lists as $list) {
+            $count += $list->count();
+        }
+        return $count;
+    }
+
+     /**
+     * Returns true if one of the SS_List's have an item.
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        foreach ($this->lists as $list) {
+            if ($list instanceof ArrayList && $list->exists()) {
+                return true;
+            }
+        }
+        foreach ($this->lists as $list) {
```

Comment:
The second loop here isn't needed - this can be an esleif inside the same loop.

It appears there's nothing to force to be an object - that should probably be caught somewhere.

Also, if the `$list` is not an `instanceof ArrayList` then there's no guarantee that `->exists()` will be a defined method.